TOOLS.rst

```diff
@@ -0,0 +1,201 @@
+  the settings belong in the ``tox`` configuration file and ``invoke``
+  will have to read them from there
+* **tasks.py** If the tool is ``invoke`` specific, then obviously the
+  configuration goes into its "configuration file" ``tasks.py``
+
+Non-tool related "settings"
+```````````````````````````
+
+**Package metadata** goes into src/ixdat/__init__.py and README.rst
+and all remaining information about how to create a package goes into
+setup.py
+
+**requirements** goes into the two requirements files;
+``requirements.txt`` for package requirements and
+``requirements-dev.txt`` for development requirements.
+
+Git hooks
+=========
+
+The version control system ``git`` has the option of having check done
+before certain important actions, like a commit or a push. These are
+referred to as ``hooks``. For xidat a pre-push is recommended (as
+opposed to a pre-commit hook, which take time to run the tests on
+every commit). The pre-push hooks are located in ``tools/hooks``.
+
+Git hooks cannot be distributed automatically, so it will need to be
+"installed".
+
+Linux instructions
+------------------
+
+On Linux systems this is done by sym-linking it into the proper
+folder. Starting at the base folder of the git archive it looks like
+this:
+
+$ cd .git/hooks
+$ ln -s ../../tools/hooks/pre-push .
+
+Windows instructions
+--------------------
+
+The pre-push hooks on Windows is dependent on a full Git installation,
+because it depends on the shell that is embedded in Git Bash. So if
+there isn't already a full Git for Windows installed, do that
+(https://git-scm.com/downloads) and afterwards confirm that the file
+``C:/Program\ Files/Git/usr/bin/sh.exe`` exists.
+
+The next step is to copy the Windows specific hook into place, so
+locate the folder ``.git/hooks`` in the main folder of the git archive
+and copy the file ``tools/hooks/pre-push_windows_git_bash`` into that
+folder. (The interested reader will notice that the only difference
+between the Linux and the Windows version of the hooks is the line at
+the top that indicates the executable that should run the program).
+
+Then there is a bit of difference depending on how git is used. If the
+main interface to git is via powershell (or mayby Command Prompt) and
+a virtual environment is being used and is active, then the commit
+hook will just work.
+
+If the development is done elsewhere, but still somehow relies on a
+virtual environment (either a separate virtual environment or an
+anaconda environment), and git is used via the Git Bash program that
+is installed along with Git for Windows, then a bit more work is
+required before. The problem is that Git Bash does not know about the
+path of the development tools, so that will have to be set manually.
+
+First we should locate the path of the tools. In case a separate
+virtual environment is used, located e.g. in c:\venv\ixdat, then
+c:\venv\ixdat\Scripts will be the path of tools. TODO anaconda.
```

Comment:
I'm using anaconda and get the following, in the PyCharm Ipython console:
```
>>> import sys
>>> sys.executable
'C:\\ProgramData\\Anaconda3\\envs\\ixdat\\python.exe'
```
so...
/c/ProgramData/Anaconda3/envs/ixdat/python.exe
?

Reply:
It's not working for me. After adding the line to ~ /.bashrc and sourcing ~/.bashrc, it's there:
```
$ echo $PATH
/c/ProgramData/Anaconda3/envs/ixdat: ....
```
but `invoke` remains unrecognized by git.
I did remember to install invoke in ixdat's python environment in PyCharm (pip install invoke in Anaconda only gets it onto its `(base)` environment).
So I think some things are missing in these instructions.

Reply:
Yeah. I'm slowly learning that while anaconda is really good for getting a lot of high quality, high speed scientific code into the hands of scientists in a hurry I find that it is actually somewhat harder to figure out exactly what goes on. I actually intended to update this later specifically for anaconda (since I just uninstalled it in favor of an ordinary python installation and virtual environments, but since you show an interest in the hooks I think I will go back to get this to work and properly documentet.

Reply:
Actually, can I persuade you do me one favor. Can I get you to run:

```cmd
echo %path%
```

in your anaconda prompt and write the results here.

Reply:
![image](https://user-images.githubusercontent.com/28803592/104467512-4d086b80-55ae-11eb-9f85-3dd6fe6879ae.png)

Reply:
Great! :|

I will see about re-installing anaconda shortly, but it will delay this PR a little further. I would prefer actually to merge it and fix it later as time allows (since it is marked with a TODO in TOOLS.rst) but I will leave that up to you.

Reply:
That's fun. That command is mentioned like ... everywhere. Grumble grumble.